_posts/2018-01-21-ITE-the-scheduler.md

```diff
@@ -0,0 +1,51 @@
+---
+title: In the Engine - The Scheduler
+description: Stuff
+author: dedmen
+layout: default
+category: ITE
+---
+
+Now dedmen just needs to get some good writing skills.
+
+
+When you [spawn](https://community.bistudio.com/wiki/spawn), [execVM](https://community.bistudio.com/wiki/execVM),[exec](https://community.bistudio.com/wiki/execVM) or [execFSM](https://community.bistudio.com/wiki/execFSM) a script it is added to a internal list of scheduled scripts.
+
+
+There is one scheduler cycle per frame (`siScr` in Engine Profiler (New article to be linked here after it's written)) which happens right after the EachFrame eventhandlers. (Maybe future article about that)
+
+At the beginning of a scheduler cycle a Timer is started, it will be used to keep track of the scheduler time limit.
+The limit is 50ms in a loading screen and 3ms outside of a loading screen. We'll see how that's used a little further below.
+
+Next up all scheduled scripts (SQF,SQS,FSM) are sorted by the time they were last executed. Meaning the script that was not processed for the longest time will land at the start of the new list.
+
+
+Now we get to the actual execution of the scripts
+
+The engine loops through our sorted list, executing the scripts in order.
+
+Each Script type is handled a little differently:
+
+- A SQS script will execute until it, is done, suspends, or after it executed for a total of 3ms. If the 3ms limit is reached it will be forced to suspend.
```

Comment:
`it, is` ? 
Dunno. Used to show the start of a list

Reply:
>A SQS script will execute until it: is done, suspends, or after it executed for a total of 3ms. If the 3ms limit is reached it will be forced to suspend.

commy